@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -736,105 +747,79 @@ export class TiltViewClipPlanesEvaluator extends TopViewClipPlanesEvaluator {
-        // Acquire camera to origin vector.
-        const cameraToOrigin = worldOrigin.sub(camera.position);
-        // Extract camera X, Y, Z orientation axes into tmp vectors array.
-        camera.matrixWorld.extractBasis(
-            this.m_tmpVectors[0],
-            this.m_tmpVectors[1],
-            this.m_tmpVectors[2]
-        );
-        const xaxis = this.m_tmpVectors[0];
-        const yaxis = this.m_tmpVectors[1];
-        // Extend the far plane by the margin along the sphere normal (radius).
-        // In order to calculate far plane distance we need to find sphere tangent
-        // line that passes thru camera position, method explanation may be found in
-        // TopViewClipPlanesEvaluator.evaluateDistanceSphericalProj() method.
+
+        // For the calculation of the far plane the problem can be simplified to 2D
+        // to calculate the worst case. The maximum view distance is assumed to be
+        // the distance from the camera to the geometric horizon plus
+        // the distance from the highest elevation to the geometric horizon.
+        // The sightline between camera and a hypothetical elevation behind the
+        // geometric horizon is a tangent of the sphere, i.e., rectangular triangles are formed.
+        // Both distances to the geometric horizon can thus be calculated with Pythagorean theorem.
+        //
+        // Another rectangular triangle is formed between look at, far plane and the sightline.
+        // Distance to far plane can thus be calculated with angle between look at and sightline
+        // and the maximum view distance.
+        //         , - ~ ~ ~ - ,
+        //     , '               ' ,          E
+        //   ,___________.___________,______. farplane for elevation
+        //  ,            .   r + e    , '  /
+        // ,             .     ,  '    ,  /
+        // ,             . O '         , / te = distance elevation to horizon
+        // ,             | .           ,/
+        //  ,            |   .  r     ,/
+        //   ,         r |      .    ,
+        //     ,_________|________, '_____ farplane for horizon
+        //       ' -_, _ | _ ,  ' / T
+        //     near      |      /
+        //             h |    / t = distance camera to horizon
+        //               |  /
+        //     look at ↑ |/  ↗ sightline to horizon
+        //               C
+
+        // Use distance to geometric horizon as base for max view distance
+        const maxViewDistance =
+            this.getDistanceToHorizon(cameraAltitude) +
+            this.getDistanceToHorizon(this.maxElevation);
+
+        // Calculate angle between look at and surface normal
+        const cameraToOrigin = this.m_tmpVectors[0]
+            .set(0, 0, 0)
+            .sub(camera.position)
+            .normalize();
+        const lookAt = this.m_tmpVectors[1];
+        camera.getWorldDirection(lookAt).normalize();
+        const cosAlpha1 = cameraToOrigin.dot(lookAt);
```

Comment:
This whole stuff could be enclosed in getCameraPitch() method, assuming that we will not need cameraToOirgin later on....
Keep in mind that we already introduced some change to TopViewEvaluator that optimize far distance whenever this angle is smaller then FOV/2, I hope to see this code also here at some point - would be nice, fingers crossed :).